### PR TITLE
Improvements and fixes to Picker, new engine example

### PR DESCRIPTION
**Pull request by mvaligursky**
new engine example (link valid after engine release):
http://playcanvas.github.io/#graphics/area-picker.html

Screen Shot 2020-10-16 at 4 47 56 PM

Fixes https://github.com/playcanvas/engine/issues/2470
Fixes https://github.com/playcanvas/engine/issues/2468

Thanks to @scarletsky for contributions!

### CODE REVIEW

File: src/scene/forward-renderer.js [hunk 1]
```diff
@@ -813,6 +813,10 @@ Object.assign(ForwardRenderer.prototype, {
         device.setScissor(x, y, w, h);
 
         if (clear) {
+            // use camera clear options if any
+            if (!options)
+                options = camera._clearOptions;
```
Comment (slimbuck) [inline src/scene/forward-renderer.js:818]: what's the difference between camera._clearOptions and all the settings used below (camera._clearColor et al)?

File: src/scene/pick.js [hunk 2]
```diff
@@ -299,6 +308,9 @@ Picker.prototype.prepare = function (camera, scene, arg) {
     // save old camera state
     this.onLayerPreRender(this.layer, sourceLayer, sourceRt);
 
+    // clear registered meshes, rendering will register them against
```
Comment (slimbuck) [inline src/scene/pick.js:311]: against->again?

Comment (mvaligursky) [reply src/scene/pick.js]: fixed, thanks!

File: src/scene/forward-renderer.js (hunk 1 above)
Comment (mvaligursky) [reply src/scene/forward-renderer.js:818]: There are two ways .. one is settings on camera, one is override settings on camera. 
Don't ask ... a lot of this code seems over engineered to me.

Comment (slimbuck) [reply src/scene/forward-renderer.js:818]: Ok I take the question back!

**Review by slimbuck — APPROVED** [review]
LGTM and also excited to see the new demo!

**Pull request merged**